core/src/main/java/io/questdb/cairo/wal/WalWriter.java

```diff
@@ -512,7 +512,10 @@ public String toString() {
     }
 
     @Override
-    public void truncate() {
+    public void truncate(boolean keepSymbolTables) {
+        if (!keepSymbolTables) {
+            throw new UnsupportedOperationException("cannot truncate symbol tables on WAL table");
```

Comment:
If we start throwing on 

```
truncate wal_table
```

this will look like regression. I'd rather avoid it and keep existing behavior until it's handled.

In any case, the error should suggest the supported syntax instead of `cannot truncate`.
Also `symbol tables`is confusing, I'd call it `symbol maps`

Reply:
> If we start throwing on
> 
> ```
> truncate wal_table
> ```

With this patch, we won't be throwing on `TRUNCATE wal_table;` SQL (check the call in `SqlCompiler#truncateTables()`). The existing behavior for SQL and embedded API is kept as is.